authentik/providers/oauth2/views/token.py

```diff
@@ -490,12 +490,13 @@ def create_code_response(self) -> dict[str, Any]:
             auth_time=self.params.authorization_code.auth_time,
             session_id=self.params.authorization_code.session_id,
         )
-        access_token.id_token = IDToken.new(
+        access_id_token = IDToken.new(
             self.provider,
             access_token,
             self.request,
         )
-        access_token.id_token.nonce = self.params.authorization_code.nonce
+        access_id_token.nonce = self.params.authorization_code.nonce
+        access_token.id_token = access_id_token
```

Comment:
We need to change the .nonce attribute before assigning to `.id_token` as that property will turn the IDToken into a dict....sigh